src/nodes.coffee

```diff
@@ -1708,11 +1737,25 @@ exports.Assign = class Assign extends Base
   unfoldSoak: (o) ->
     unfoldSoak o, this, 'variable'
 
+  objectHasSplat: (o, obj) ->
```

Comment:
Yes, indeed. These checkings are all over the place and should be tamed.
The truth is, I just started to use CS and I've kinda rushed in without proper knowledge of the CS internals.